Firestore/core/src/firebase/firestore/api/query_core.h

```diff
@@ -164,6 +164,23 @@ class Query {
   void ValidateNewOrderByPath(const model::FieldPath& fieldPath) const;
   void ValidateOrderByField(const model::FieldPath& orderByField,
                             const model::FieldPath& inequalityField) const;
+  /**
+   * Validates that the value passed into a disjunctive filter satisfies all
+   * array requirements.
+   */
+  void ValidateDisjunctiveFilterElements(const model::FieldValue& field_value,
+                                         const core::Filter::Operator op) const;
```

Comment:
Filter::Operator is an enum. There's no reason to mark it `const` when passing it, and indeed it's slightly misleading to do so because `const Operator` and `Operator` are the same for the purpose of overload resolution.